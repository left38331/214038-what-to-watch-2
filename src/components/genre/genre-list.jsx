import React from 'react';
import PropTypes from 'prop-types';

export const GenreList = (props) => {
  return <ul className="catalog__genres-list">
    {props.genresList.map((item, i) => <li className={`catalog__genres-item ${item === props.activeGenre ? `catalog__genres-item--active` : ``} `} key={i + item}>
      <a href="#" className="catalog__genres-link" onClick={()=> {
        props.onChangeGenre(item);
      }}>{item}</a>
    </li>)}
  </ul>;
};

GenreList.propTypes = {
  genresList: PropTypes.array.isRequired,
  activeGenre: PropTypes.string.isRequired,
  onChangeGenre: PropTypes.func.isRequired,
};
