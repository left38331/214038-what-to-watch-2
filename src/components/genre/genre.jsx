import React from 'react';
import PropTypes from 'prop-types';

export const Genre = (props) => {
  const {genre} = props;

  return <li className="catalog__genres-item">
    <a href="#" className="catalog__genres-link">{genre}</a>
  </li>;
};

Genre.propTypes = {
  genre: PropTypes.oneOf([`Comedies`, `Crime`, `Documentary`, `Dramas`, `Horror`, `Kids & Family`, `Romance`, `Sci-Fi`, `Thrillers`])
};
