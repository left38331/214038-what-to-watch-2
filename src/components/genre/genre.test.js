import React from 'react';
import renderer from 'react-test-renderer';
import {Genre} from "./genre";

it(`Render correctly genre component`, () => {
  const genreComponent = renderer.create(<Genre
    genre = {`Comedies`}
  />).toJSON();

  expect(genreComponent).toMatchSnapshot();
});
