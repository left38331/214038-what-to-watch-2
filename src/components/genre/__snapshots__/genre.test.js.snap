// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render correctly genre component 1`] = `
<li
  className="catalog__genres-item"
>
  <a
    className="catalog__genres-link"
    href="#"
  >
    Comedies
  </a>
</li>
`;
